### PR TITLE
Added aten::triangular_solve op

### PR DESCRIPTION
**Pull request by vishwakftw**

### CODE REVIEW

File: torch_xla/csrc/aten_xla_type.cpp [hunk 1]
```diff
@@ -2392,6 +2392,18 @@ at::Tensor& AtenXlaType::transpose_(at::Tensor& self, int64_t dim0,
   return self;
 }
 
+std::tuple<at::Tensor, at::Tensor> AtenXlaType::triangular_solve(
+    const at::Tensor& b, const at::Tensor& A, bool upper, bool transpose,
+    bool unitriangular) const {
+  // Currently, ATen doesn't have a left_side option. Once this
+  // is added, this API will have to be changed.
+  auto results = XLATensor::triangular_solve(bridge::GetXlaTensor(b),
+                                             bridge::GetXlaTensor(A), true,
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: ```/*left_side=*/true```

File: torch_xla/csrc/ops/triangular_solve.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+// This function plays two roles:
+// - computes the output shape
+// - computes the broadcasted shape for the operands
+// NB: This currently infers the shape when left_side is true, as done in ATen
+std::pair<xla::Shape, xla::Shape> InferTriangularSolveShape(
+    const xla::Shape& rhs_shape, const xla::Shape& lhs_shape) {
+  // Obtain the number of right-hand sides, and dimension of the square matrix
+  xla::int64 nrhs = rhs_shape.dimensions(rhs_shape.rank() - 1);
+  xla::int64 n = lhs_shape.dimensions(lhs_shape.rank() - 1);
+
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/triangular_solve.cpp]: line--

File: torch_xla/csrc/ops/triangular_solve.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  xla::Shape rhs_batch_shape(rhs_shape);
+  xla::Shape lhs_batch_shape(lhs_shape);
+
+  rhs_batch_shape.DeleteDimension(rhs_batch_shape.rank() - 1);
+  lhs_batch_shape.DeleteDimension(lhs_batch_shape.rank() - 1);
+
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/triangular_solve.cpp]: line--

File: torch_xla/csrc/ops/triangular_solve.cpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+#include "torch_xla/csrc/helpers.h"
+#include "torch_xla/csrc/lowering_context.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+// This function plays two roles:
+// - computes the output shape
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/triangular_solve.cpp]: Computes ...

Sentences begin with Capital, and ends with dot (.).

File: torch_xla/csrc/ops/triangular_solve.cpp [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+// - computes the broadcasted shape for the operands
+// NB: This currently infers the shape when left_side is true, as done in ATen
```
Comment (dlibenzi) [inline torch_xla/csrc/ops/triangular_solve.cpp]: Ends with dot. Recheck your comments.

File: torch_xla/csrc/aten_xla_type.cpp (hunk 1 above)
Comment (vishwakftw) [reply torch_xla/csrc/aten_xla_type.cpp]: Thank you, I've addressed this.

File: torch_xla/csrc/ops/triangular_solve.cpp (hunk 2 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/triangular_solve.cpp]: Done.

File: torch_xla/csrc/ops/triangular_solve.cpp (hunk 3 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/triangular_solve.cpp]: Done.

File: torch_xla/csrc/ops/triangular_solve.cpp (hunk 4 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/triangular_solve.cpp]: Sorry about that. I will keep this in mind henceforth.

File: torch_xla/csrc/ops/triangular_solve.cpp (hunk 5 above)
Comment (vishwakftw) [reply torch_xla/csrc/ops/triangular_solve.cpp]: Sure, done.

**Comment by vishwakftw** [comment]
I ran clang-format with Google style (`clang-format -style=Google`) for formatting, as suggested.

**Pull request merged**